Add installation instructions of MongoDB for non-windows OS & Fix mistakes in frontend README instructions

ysway opened the pull request:
Closes #230
Closes #233

## Proposed Changes
- Add installation instructions of MongoDB for non-windows OS
- Fix mistakes in frontend README instructions

ysway committed:
Fix habit views

ysway committed:
Merge branch 'main' of github.com:UOA-SE701-Group3-2021/3Lancers into main

ysway committed:
Merge branch 'main' of github.com:UOA-SE701-Group3-2021/3Lancers into main

ysway committed:
Merge branch 'main' of github.com:UOA-SE701-Group3-2021/3Lancers into main

ysway committed:
Add a unique key for each child in a list and add code comments

Co-authored-by: lhsunyc <ysun499@aucklanduni.ac.nz>
Signed-off-by: Siwei Yang <syan553@aucklanduni.ac.nz>

ysway committed:
Refactoring code and merge duplicated CSS

Co-authored-by: lhsunyc <ysun499@aucklanduni.ac.nz>
Signed-off-by: Siwei Yang <syan553@aucklanduni.ac.nz>

ysway committed:
Add mssing line at EOF to HabitTracker.module.css

Co-authored-by: lhsunyc <ysun499@aucklanduni.ac.nz>
Signed-off-by: Siwei Yang <syan553@aucklanduni.ac.nz>

ysway committed:
Merge branch 'main' of github.com:UOA-SE701-Group3-2021/3Lancers into main

ysway committed:
Add installation instructions of MongoDB 

Add installation instructions of MongoDB for non-windows OS


Signed-off-by: Siwei Yang <syan553@aucklanduni.ac.nz>

maheknangia reviewed (approved):
This looks a lot better Siwei! Very easy to follow through and covers the operating systems. Thanks for this!

ysway committed:
Fix mistakes in frontend README instructions

- Working directory changes to telos-frontend
- Removal of npm run lint from instructions


Co-authored-by: ysway <yswysc@gmail.com>
Signed-off-by: ysway <yswysc@gmail.com>
Signed-off-by: Siwei Yang <syan553@aucklanduni.ac.nz>

cng822 reviewed (approved):
LGTM! Thanks for doing this. Makes it so much easier now for different OS :)

milkyway404 reviewed (approved):
Looks awesome, thank you Siwei!
